python/pyspark/sql/profiler.py

```diff
@@ -239,3 +241,72 @@ def _profile_results(self) -> "ProfileResults":
         with self._lock:
             value = self._accumulator.value
             return value if value is not None else {}
+
+
+class Profile:
+    """User-facing profile API. This instance can be accessed by
```

Comment:
We should probably also fix the docs at `python/docs/source/reference`.

Reply:
Good point! Would you mind if I file a follow-up pr for documentation? We might also want to edit the PySpark debugging guide together.